test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -212,34 +218,242 @@ func (s *snapshottableTestSuite) DefineTests(driver TestDriver, pattern testpatt
+			if err != nil && !apierrors.IsNotFound(err) {
+				framework.Failf("Error deleting snapshot %q. Error: %v", pvc.Name, err)
+			}
+		}()
+		err = WaitForSnapshotReady(dc, snapshot.GetNamespace(), snapshot.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("checking the snapshot")
+		// Get new copy of the snapshot
+		snapshot, err = dc.Resource(SnapshotGVR).Namespace(snapshot.GetNamespace()).Get(context.TODO(), snapshot.GetName(), metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		// Get the bound snapshotContent
+		snapshotStatus := snapshot.Object["status"].(map[string]interface{})
+		snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+		snapshotContent, err := dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("checking the SnapshotContent")
+		snapshotContentSpec := snapshotContent.Object["spec"].(map[string]interface{})
+		volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+		framework.ExpectEqual(snapshotContentSpec["volumeSnapshotClassName"], vsc.GetName())
+		framework.ExpectEqual(snapshotContentSpec["deletionPolicy"], "Retain")
+		framework.ExpectEqual(volumeSnapshotRef["name"], snapshot.GetName())
+		framework.ExpectEqual(volumeSnapshotRef["namespace"], snapshot.GetNamespace())
+
+		ginkgo.By("deleting the snapshot")
+		err = dc.Resource(SnapshotGVR).Namespace(snapshot.GetNamespace()).Delete(context.TODO(), snapshot.GetName(), metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			framework.Failf("Error deleting snapshot %q. Error: %v", pvc.Name, err)
+		}
+
+		ginkgo.By("checking the SnapshotContent has not been deleted")
+		_, err = dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		if strings.Contains(driver.GetDriverInfo().Name, "hostpath") {
+			ginkgo.By("manually deleting the SnapshotContent")
+			err = dc.Resource(SnapshotContentGVR).Namespace(snapshotContent.GetNamespace()).Delete(context.TODO(), snapshotContent.GetName(), metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				framework.Failf("Error deleting snapshot %q. Error: %v", pvc.Name, err)
+			}
+
+			ginkgo.By("checking the physical snapshot still remains")
+			nodeName := config.ClientNodeSelection.Name
+			if nodeName == "" {
+				// for tests where ClientNodeSelection.Name is empty, manually find hostpath node
+				nodeName, err = getHostpathNodeName(cs, "default")
+				framework.ExpectNoError(err)
+			}
+			node, err := cs.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			hostExec := utils.NewHostExec(f)
+			_, err = hostExec.Execute(command, node)
```

Comment:
This is checking that you can still write to the volume with this command  `echo 'hello world' > /mnt/test/data`, but not checking if the physical snapshot still exists?
Can you check if the physical snapshot exists before deleting the VolumeSnapshot and then verify it is still there after deleting?

Reply:
Yes, I originally had a command to do this, but I accidentally deleted it in a refactor. Fixing and testing now

Reply:
Fixed this, the test is now finding the snapshot at `/var/lib/csi-hostpath-data/<snapshotHandle>.snap`